flink-table/flink-sql-gateway/src/main/java/org/apache/flink/table/gateway/service/operation/OperationExecutor.java

```diff
@@ -292,10 +292,16 @@ public ResultFetcher getCompletionHints(
     @VisibleForTesting
     public TableEnvironmentInternal getTableEnvironment() {
         TableEnvironmentInternal tableEnv = sessionContext.createTableEnvironment();
-        tableEnv.getConfig().getConfiguration().addAll(executionConfig);
+        tableEnv.getConfig().getConfiguration().addAll(operationConfig);
```

Comment:
After rethink, it's better we can give the execution config when creating the table env. Because we can not modify the configuration after creating envrionment, e.g. execution.runtime-mode